### PR TITLE
Documented p_misc.[ch]

### PR DESCRIPTION
**Pull request by wyld-sw**

### CODE REVIEW

File: src/p_misc.c [hunk 1]
```diff
@@ -213,151 +393,282 @@ prim_fmttime(PRIM_PROTOTYPE)
...
     if (oper1->type != PROG_STRING)
-	abort_interp("Non-string argument.");
+        abort_interp("Non-string argument.");
+
     if (mlev < tp_userlog_mlev)
-	abort_interp("Permission Denied (mlev < tp_userlog_mlev)");
+        abort_interp("Permission Denied (mlev < tp_userlog_mlev)");
+
+    /**
+     * @TODO buf = DoNullInd(oper1->data.string);
```
Comment (tanabi) [inline src/p_misc.c]: This is a pretty generic @TODO ... is the purpose to add the line that you have here?  Is there something else to do here?

File: src/p_misc.c [hunk 2]
```diff
@@ -476,118 +810,195 @@ prim_fork(PRIM_PROTOTYPE)
...
+        /* tmp, ref */
+        rooms = exits = things = players = programs = garbage = 0;
+
+        for (dbref i = 0; i < db_top; i++) {
+            if (ref == NOTHING || OWNER(i) == ref) {
+                switch (Typeof(i)) {
+                    case TYPE_ROOM:
+                        rooms++;
+                        break;
+                    case TYPE_EXIT:
```
Comment (tanabi) [inline src/p_misc.c]: To be technically correct (the best kind of correct) there should be a blank line between each 'case' block.  So like:

```
switch(Typeof(i)) {
    case TYPE_ROOM:
        rooms++;
        break;

    case TYPE_EXIT: /* and so forth */
```

File: src/p_misc.c [hunk 3]
```diff
@@ -476,118 +810,195 @@ prim_fork(PRIM_PROTOTYPE)
...
-		garbage++;
-		break;
-	    }
-	}
+        if (ref == NOTHING || OWNER(i) == ref) {
+            switch (Typeof(i)) {
+                case TYPE_ROOM:
+                    rooms++;
+                    break;
+                case TYPE_EXIT:
```
Comment (tanabi) [inline src/p_misc.c]: Same note as above re. switch spacing.

File: src/p_misc.c [hunk 4]
```diff
@@ -821,115 +1429,217 @@ prim_setsysparm(PRIM_PROTOTYPE)
...
-    case TUNESET_BADVAL:
-        free(oldvalue);
-	abort_interp("Bad parameter value. (2)");
-    case TUNESET_DENIED:
-        free(oldvalue);
-	abort_interp("Permission denied. (1)");
-    default:
-        free(oldvalue);
-	break;
+        case TUNESET_SUCCESS:
```
Comment (tanabi) [inline src/p_misc.c]: Same note as above re. case spacing.

File: src/p_misc.c [hunk 5]
```diff
@@ -992,76 +1732,130 @@ prim_ext_name_okp(PRIM_PROTOTYPE)
...
-	for (ref = 0; buf[ref]; ref++)
-	    buf[ref] = tolower(buf[ref]);
-	if (!strcmp(buf, "e") || !strcmp(buf, "exit")) {
+        strcpyn(buf, sizeof(buf), DoNullInd(oper2->data.string));
+
+        for (ref = 0; buf[ref]; ref++)
+            buf[ref] = tolower(buf[ref]);
+
+        /**
+         * @TODO Maybe these should use case-insenstive?
```
Comment (tanabi) [inline src/p_misc.c]: Agreed.  I'd take it a step further and strncmp them based on the length of the input string.

So for example, "ex" would match "exit".  But doing case insensitive is trivial so we should definitely do that :D

File: src/p_misc.c (hunk 1 above)
Comment (wyld-sw) [reply src/p_misc.c]: Changed to say that it may be able to replace the "if" construct entirely.

File: src/p_misc.c (hunk 2 above)
Comment (wyld-sw) [reply src/p_misc.c]: Done.

File: src/p_misc.c (hunk 3 above)
Comment (wyld-sw) [reply src/p_misc.c]: Done.

File: src/p_misc.c (hunk 4 above)
Comment (wyld-sw) [reply src/p_misc.c]: Done.

File: src/p_misc.c (hunk 5 above)
Comment (wyld-sw) [reply src/p_misc.c]: Might need a special case for "p" - does it match "player" or "program"? Also, should "m" and "mu" also be aliases for "muf"?

I'm sure it's doable.

**Pull request merged**